appstudio-controller/controllers/appstudio.redhat.com/deploymenttargetreclaimer_controller.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package appstudioredhatcom
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	codereadytoolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	applicationv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// DeploymentTargetReconciler reconciles a DeploymentTarget object
+type DeploymentTargetReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+const (
+	FinalizerDT = "dt.appstudio.redhat.com/finalizer"
+)
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims/finalizers,verbs=update
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets,verbs=get;list;watch;update;patch;delete
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets/finalizers,verbs=update
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacerequest,verbs=get;list;watch;create;update;patch;delete
```

Comment:
The resource needs to be plural (`spacerequests`), I had made the same mistake and it had to be fixed in #427

Reply:
Updated

Reply:
Ack, thanks!